From 296e5aba4798d8dbf2d949bdebbf70496c312c1e Mon Sep 17 00:00:00 2001
From: Daniel Borkmann <dborkman@redhat.com>
Date: Tue, 22 Jan 2013 14:04:34 +0100
Subject: [PATCH 39/53] apps: sctp_darn: fix format string warning on some
 archs like ppc64
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:

  sctp_darn.c: In function ‘get_assocstats_func’:
  sctp_darn.c:2261: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2262: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2263: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2264: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2265: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2266: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2267: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2268: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2269: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2270: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2271: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2272: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2273: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2274: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’
  sctp_darn.c:2275: warning: format ‘%llu’ expects type ‘long long unsigned int’, but argument 2 has type ‘__u64’

We use the generic PRIu64 type instead for printing.

Signed-off-by: Daniel Borkmann <dborkman@redhat.com>
---
 src/apps/sctp_darn.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/apps/sctp_darn.c b/src/apps/sctp_darn.c
index 551e49d..9970cfb 100644
--- a/src/apps/sctp_darn.c
+++ b/src/apps/sctp_darn.c
@@ -54,7 +54,7 @@
 //#define _GNU_SOURCE
 #include <getopt.h>
 #include <netdb.h>
-
+#include <inttypes.h>
 #include <ctype.h>
 #include <sys/types.h>
 #include <sys/socket.h>
@@ -2258,21 +2258,21 @@ get_assocstats_func(int sk, sctp_assoc_t assoc_id)
 		return error;
 	}
 
-	printf("Retransmitted Chunks: %llu\n",              stats.sas_rtxchunks);
-	printf("Gap Acknowledgements Received: %llu\n",     stats.sas_gapcnt);
-	printf("TSN received > next expected: %llu\n",      stats.sas_outofseqtsns);
-	printf("SACKs sent: %llu\n",                        stats.sas_osacks);
-	printf("SACKs received: %llu\n",                    stats.sas_isacks);
-	printf("Control chunks sent : %llu\n",              stats.sas_octrlchunks);
-	printf("Control chunks received : %llu\n",          stats.sas_ictrlchunks);
-	printf("Ordered data chunks sent: %llu\n",          stats.sas_oodchunks);
-	printf("Ordered data chunks received: %llu\n",      stats.sas_iodchunks);
-	printf("Unordered data chunks sent: %llu\n",        stats.sas_ouodchunks);
-	printf("Unordered data chunks received: %llu\n",    stats.sas_iuodchunks);
-	printf("Dups received (ordered+unordered): %llu\n", stats.sas_idupchunks);
-	printf("Packets sent: %llu\n",                      stats.sas_opackets);
-	printf("Packets received: %llu\n",                  stats.sas_ipackets);
-	printf("Maximum Observed RTO this period: %llu - Transport: ", stats.sas_maxrto);
+	printf("Retransmitted Chunks: %" PRIu64 "\n", stats.sas_rtxchunks);
+	printf("Gap Acknowledgements Received: %" PRIu64 "\n", stats.sas_gapcnt);
+	printf("TSN received > next expected: %" PRIu64 "\n", stats.sas_outofseqtsns);
+	printf("SACKs sent: %" PRIu64 "\n", stats.sas_osacks);
+	printf("SACKs received: %" PRIu64 "\n", stats.sas_isacks);
+	printf("Control chunks sent: %" PRIu64 "\n", stats.sas_octrlchunks);
+	printf("Control chunks received: %" PRIu64 "\n", stats.sas_ictrlchunks);
+	printf("Ordered data chunks sent: %" PRIu64 "\n", stats.sas_oodchunks);
+	printf("Ordered data chunks received: %" PRIu64 "\n", stats.sas_iodchunks);
+	printf("Unordered data chunks sent: %" PRIu64 "\n", stats.sas_ouodchunks);
+	printf("Unordered data chunks received: %" PRIu64 "\n", stats.sas_iuodchunks);
+	printf("Dups received (ordered+unordered): %" PRIu64 "\n", stats.sas_idupchunks);
+	printf("Packets sent: %" PRIu64 "\n", stats.sas_opackets);
+	printf("Packets received: %" PRIu64 "\n", stats.sas_ipackets);
+	printf("Maximum Observed RTO this period: %" PRIu64 " - Transport: ", stats.sas_maxrto);
 	print_sockaddr((struct sockaddr *)&stats.sas_obs_rto_ipaddr);
 	printf("\n");
 
-- 
1.7.11.7

